lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -12,15 +12,22 @@
 #along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 require 'chef/handler'
+require 'digest/md5'
 
 module ChefHandlerForeman
   class ForemanFacts < Chef::Handler
     attr_accessor :uploader
+    attr_accessor :blacklist
+    attr_accessor :whitelist
+    attr_accessor :cachefile
```

Comment:
last nitpick, could you please rename `cachefile` to `cache_file` throughout the PR? sounds more like two words to me and it would be more consistent with `cache_expired` name.